app/src/main/java/org/p2p/wallet/home/addmoney/repository/AddMoneyCellsRepository.kt

```diff
@@ -0,0 +1,74 @@
+package org.p2p.wallet.home.addmoney.repository
+
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.wallet.R
+import org.p2p.wallet.common.InAppFeatureFlags
+import org.p2p.wallet.common.feature_toggles.toggles.remote.StrigaSignupEnabledFeatureToggle
+import org.p2p.wallet.home.addmoney.mapper.AddMoneyCellMapper
+import org.p2p.wallet.home.addmoney.model.AddMoneyItemType
+import org.p2p.wallet.infrastructure.network.provider.SeedPhraseProvider
+import org.p2p.wallet.settings.interactor.SettingsInteractor
+import org.p2p.wallet.striga.signup.steps.interactor.StrigaSignupInteractor
+
+class AddMoneyCellsRepository(
```

Comment:
Migrated from hardcoded views in xml to cells, now items can be added or removed dynamically

Reply:
looks good!

Reply:
is it okay to manipulate UI models in :data layer? I would instead return some domain models and map them in the presenter. It will be more explicit to return not AnyCellItem

Reply:
@gslevinkov ui-model is also model and contains data
anyway, look at the new implementation, I've added data class model + enum with constants + mapping to ui-model